test/run-test.ps1

```diff
@@ -1,169 +1,62 @@
-
-        $dotnetNewParam = "console --framework netcoreapp$netcoreappVersion"
+if ($LASTEXITCODE -ne 0) { throw "Failed to install the .NET Core SDK" }
 
-        $optionalRestoreParams = ""
-        if ($sdkTag -like $includePrereleasePackageSourceForSdkTag) {
-            $optionalRestoreParams = "-s https://dotnet.myget.org/F/dotnet-core/api/v3/index.json -s https://api.nuget.org/v3/index.json"
-        }
+Push-Location "$PSScriptRoot\Microsoft.DotNet.Docker.Tests"
 
-        Write-Host "----- Testing create, restore and build with $fullSdkTag with image $buildImage -----"
-        Try {
-            exec { (Get-Content ${testFilesPath}Dockerfile.test).
-                    Replace("{image}", $fullSdkTag).
-                    Replace("{dotnetNewParam}", $dotnetNewParam).
-                    Replace("{optionalRestoreParams}", $optionalRestoreParams) `
-                | docker build $optionalDockerBuildArgs -t $buildImage -
-            }
-
-            Write-Host "----- Running app built on $fullSdkTag -----"
-            exec { docker run --rm $buildImage dotnet run }
-
-            $framworkDepVol = "framework-dep-publish-$appName"
-            Write-Host "----- Publishing framework-dependant app built on $fullSdkTag to volume $framworkDepVol -----"
-            Try {
-                exec { docker run --rm `
-                    -v ${framworkDepVol}:"${containerRoot}volume" `
-                    $buildImage `
-                    dotnet publish -o ${containerRoot}volume
-                }
-
-                $fullRuntimeTag = Get-RuntimeTag $_.Dockerfile "runtime" $activeOS $manifestRepo
-                Write-Host "----- Testing on $fullRuntimeTag with $sdkTag framework-dependent app -----"
-                exec { docker run --rm `
-                    -v ${framworkDepVol}":${containerRoot}volume" `
-                    "$fullRuntimeTag" `
-                    dotnet "${containerRoot}volume${platformDirSeparator}test.dll"
-                }
-            }
-            Finally {
-                docker volume rm $framworkDepVol
-            }
-
-            if ($activeOS -eq "linux") {
-                $selfContainedImage = "self-contained-build-${buildImage}"
-                Write-Host "----- Creating publish-image for self-contained app built on $fullSdkTag -----"
-                Try {
-                    if ($sdkTag -like "2.0-sdk*")
-                    {
-                        # Workaround for https://github.com/dotnet/sdk/issues/1570
-                        $optionalRestoreParams = " /p:runtimeidentifier=debian.8-x64"
-                    }
-
-                    exec { (Get-Content ${testFilesPath}Dockerfile.linux.publish).
-                                Replace("{image}", $buildImage).
-                                Replace("{optionalRestoreParams}", $optionalRestoreParams) `
-                        | docker build $optionalDockerBuildArgs -t $selfContainedImage -
-                    }
+Try {
+    # Run Tests
```

Comment:
Minor - Precondition to run tests is `Dockerfile.test` is available. If in the unlikely case it isn't available, better to display error and exit.

Reply:
The Dockerfile.test gets copied to the output dir as part of the build.  Since dotnet test performs the build, I'm not sure how you would check the precondition.  Did I misunderstand the comment?

Reply:
Please ignore my comment. I misunderstood how Dockerfile.test is used in build.